src/azure_pdf_parser/utils.py

```diff
@@ -29,74 +32,125 @@ def call_api_with_error_handling(retries: int, func, *args, **kwargs) -> Any:
-    return page
+                    bounding_region.page_number = (
+                        bounding_region.page_number + batch.page_range[0] - 1
+                    )
+    return batch
 
 
-def merge_responses(pages: Sequence[PDFPage]) -> AnalyzeResult:
+def merge_responses(batches: Sequence[PDFPagesBatchExtracted]) -> AnalyzeResult:
     """
-    Merge individual page responses from multiple API calls into one.
+    Merge page batch responses from multiple API calls into one.
 
     Currently, merging is done by concatenating the paragraphs and tables from each
     page.
 
     This is as styles, documents, languages is found to be empty and we are only
     concerned with the tables and paragraphs for CPR purposes. If this changes,
     we will be storing the raw api responses and thus will be able to recover state.
+
+    Note that the content field is not required to be appended to in the merge analyse
+    result as this content duplicates the data in the paragraphs.
     """
-    pages = [propagate_page_number(page) for page in pages]
+    batches = [propagate_page_number(batch) for batch in batches]
 
     all_paragraphs = []
     all_tables = []
-    for page in pages:
-        if page.extracted_content.paragraphs:
-            all_paragraphs.extend(page.extracted_content.paragraphs)
-        if page.extracted_content.tables:
-            all_tables.extend(page.extracted_content.tables)
-
-    merged_analyse_result: AnalyzeResult = pages.pop(0).extracted_content
+    for batch in batches:
+        if batch.extracted_content.paragraphs:
+            all_paragraphs.extend(batch.extracted_content.paragraphs)
+        if batch.extracted_content.tables:
+            all_tables.extend(batch.extracted_content.tables)
+
+    # Copy the first result to a variable and add the content for all the pages.
+    merged_analyse_result: AnalyzeResult = batches.pop(0).extracted_content
     merged_analyse_result.paragraphs = all_paragraphs
     merged_analyse_result.tables = all_tables
 
     return merged_analyse_result
 
 
-def split_into_pages(document_bytes: BytesIO) -> dict[int, bytes]:
-    """Split the API response into individual pages."""
+def split_into_batches(
+    document_bytes: BytesIO, batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+) -> list[PDFPagesBatch]:
+    if not batch_size:
+        batch_size = DEFAULT_BATCH_SIZE
+
+    if batch_size < 1:
+        raise ValueError("Batch size must be greater than 0.")
+
+    """Split the API response into a batch of pages."""
+    logger.info(
+        "Splitting pdf into batches.", extra={"props": {"batch size": batch_size}}
+    )
     pdf = PdfReader(document_bytes)
 
-    pages_dict = {}
-    for page_number, page in enumerate(pdf.pages):
+    batches: list[list] = [
+        pdf.pages[page_index : page_index + batch_size]
+        for batch, page_index in enumerate(range(0, len(pdf.pages), batch_size))
+    ]
+
+    batches_with_bytes = []
+    for index, batch in enumerate(batches):
```

Comment:
if you accept the above suggestion we can avoid the double enumeration & just do it here)...
```suggestion
    for batch_index, pages in enumerate(page_batches):
```